File: testprojects/csharp/benchmark/domain/google_plus/Observable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using CSharpBenchmark.presentation.gui.observer.google_plus;

namespace CSharpBenchmark.domain.google_plus
{

    public abstract class Observable
    {
        private List<presentation.gui.observer.google_plus.Observer> observers = new List<Observer>();

        public void attach(presentation.gui.observer.google_plus.Observer o)
        {
            observers.Add(o);
        }

        public void detach(Observer o)
        {
            observers.Remove(o);
        }

        public void notifyObservers()
        {
            foreach (Observer o in observers)
            {
                o.update();
            }
        }
    }
}

File: testresources/java/benchmark_2012/domain/google_plus/Observable.java
package domain.google_plus;

import java.util.ArrayList;
import java.util.List;

import presentation.gui.observer.google_plus.Observer;

public abstract class Observable {
	private List<presentation.gui.observer.google_plus.Observer> observers = new ArrayList<Observer>();

	public void attach(presentation.gui.observer.google_plus.Observer o){
		observers.add(o);
	}

	public void detach(Observer o){
		observers.remove(o);
	}	

	public void notifyObservers(){
		for(Observer o : observers){
			o.update();
		}
	}
}